### PR TITLE
id-67: ability to match on non-specific array item

### PR DESCRIPTION
**Pull request by iw-ideinko**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -342,9 +342,22 @@ internals.addMethod(['instanceof', 'instanceOf'], internals.instanceof);
 
 internals.match = function (regex) {
 
+    if (this._ref instanceof Array) {
```
Comment (cjihrig) [inline lib/index.js]: This check should use `Array.isArray()`.

File: lib/index.js [hunk 2]
```diff
@@ -342,9 +342,22 @@ internals.addMethod(['instanceof', 'instanceOf'], internals.instanceof);
...
+
```
Comment (cjihrig) [inline lib/index.js]: No need for this blank line.

File: lib/index.js [hunk 3]
```diff
@@ -342,9 +342,22 @@ internals.addMethod(['instanceof', 'instanceOf'], internals.instanceof);
...
+        return this.assert(internals.matchArray(this._ref, regex), 'match ' + regex);
+    }
     return this.assert(regex.exec(this._ref), 'match ' + regex);
```
Comment (cjihrig) [inline lib/index.js]: Blank line before this please.

File: lib/index.js [hunk 4]
```diff
@@ -342,9 +342,22 @@ internals.addMethod(['instanceof', 'instanceOf'], internals.instanceof);
...
 };
 
+internals.matchArray = function (items, regex) {
```
Comment (cjihrig) [inline lib/index.js]: I don't think this needs to be a separate function. You can just do the `some()` in `match()`

File: lib/index.js [hunk 5]
```diff
@@ -342,9 +342,22 @@ internals.addMethod(['instanceof', 'instanceOf'], internals.instanceof);
...
+
+    return items.some((item) =>
+
+        regex.exec(item) !== null
```
Comment (cjihrig) [inline lib/index.js]: Either put this on the same line as the `=>` or wrap the body in curly braces.

Comment (cjihrig) [inline lib/index.js]: Missing semicolon.

File: lib/index.js [hunk 6]
```diff
@@ -342,9 +342,22 @@ internals.addMethod(['instanceof', 'instanceOf'], internals.instanceof);
...
+    );
+
```
Comment (cjihrig) [inline lib/index.js]: Unnecessary blank line.

File: test/index.js [hunk 7]
```diff
@@ -1908,6 +1908,26 @@ describe('expect()', () => {
                 Hoek.assert(exception.message === 'Expected \'a4x\' to match /\\w\\dy/', exception);
                 done();
             });
+
+            it('validates arrays', (done) => {
```
Comment (cjihrig) [inline test/index.js]: Please split this into two separate tests.

**Comment by cjihrig** [comment]
I think this would have to use a flag like `deep`. Otherwise, there is no way to match the entire array (not sure how common that is), and it's a breaking change.

**Comment by cjihrig** [comment]
@danielb2 care to weigh in here, since you're the one that requested #67

**Comment by danielb2** [comment]
👍

**Comment by danielb2** [comment]
This is what I was expecting. Thanks @iw-ideinko

File: test/index.js [hunk 8]
```diff
@@ -1908,6 +1908,39 @@ describe('expect()', () => {
...
+            it('validates array assertion', (done) => {
+
+                const throws = function () {
+
+                    Code.expect(['zero', 'audience']).to.match(/roa/);
+                };
+
+                let exception = false;
+                try {
+                    Code.expect(throws).to.throw();
```
Comment (cjihrig) [inline test/index.js]: This should just be `Code.expect(['zero', 'audience']).to.match(/roa/);`. Expecting it to throw makes the `try...catch` unnecessary.

File: lib/index.js [hunk 9]
```diff
@@ -342,6 +342,13 @@ internals.addMethod(['instanceof', 'instanceOf'], internals.instanceof);
 
 internals.match = function (regex) {
 
+    if (Array.isArray(this._ref)) {
```
Comment (cjihrig) [inline lib/index.js]: This needs to depend on the `deep` flag as well.

**Comment by cjihrig** [comment]
Please leave a comment when you update your PR, as GitHub doesn't notify anyone when new commits are pushed.

**Comment by cjihrig** [comment]
@iw-ideinko ping

**Comment by cjihrig** [comment]
Closing due to inactivity.

**Pull request closed without merging**